mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -368,9 +368,9 @@ class Word2Vec extends Serializable with Logging {
             var wc = wordCount
             if (wordCount - lastWordCount > 10000) {
               lwc = wordCount
-              alpha =
-                learningRate *
-                  (1 - numPartitions * wordCount.toDouble / (numIterations * trainWordsCount + 1))
+              alpha = learningRate *
+                (1 - numPartitions * wordCount.toDouble + (k - 1) * trainWordsCount /
```

Comment:
you need `numPartitions * wordCount.toDouble + (k - 1) * trainWordsCount` between parentheses 

`alpha = learningRate * (1 - (numPartitions * wordCount.toDouble + (k - 1) * trainWordsCount) / (numIterations * trainWordsCount + 1))`

Reply:
oh... Thanks! I fixed it